moveit_core/CMakeLists.txt

```diff
@@ -226,6 +231,32 @@ add_subdirectory(collision_distance_field)
 add_subdirectory(kinematics_metrics)
 add_subdirectory(dynamics_solver)
 
+set(pymoveit_libs
+  moveit_collision_detection
+  moveit_kinematic_constraints
+  moveit_planning_scene
+  moveit_python
+  moveit_robot_model
+  moveit_robot_state
+  moveit_transforms
+)
+
+if(pybind11_catkin_FOUND)
+  add_subdirectory(python)
+
+  pybind_add_module(pymoveit
```

Comment:
I'm not convinced yet about this structuring: Shouldn't we have different libs/modules for moveit_core, moveit_ros (MGI), etc? I'm missing the name `moveit_core` here.

Reply:
I sort of agree, but I'm not sure. I want to go with whatever is the most obvious/expected thing for a user.

C++ users would expect to be able to translate includes, for example. `<moveit/core/kinematic_constraints/utils.h>`. how should this translate to an import statement? I'd say it should probably by `from moveit.core.kinematic_constraints import utils` ?
Then let's take move group, you might turn `<moveit/move_group/move_group_capability.h>` into `from moveit.move_group import move_group_capability`?

But there's also an argument for using the package name, i.e `moveit_core`. ROS python users would certainly expect that, so something like `from moveit_core.kinematic_constraints import utils` and `from moveit_ros.move_group import move_group_capability`.

Either approach is ok I think, but we do need to figure out how we're going to do documentation generation for these bindings.

Reply:
also fully acknowledging I did neither of these approaches here, what I did is definitely bad. What do folks think is the best structure?

Reply:
Another possibly important note here -- if we ever want to add pure python to the package (whatever it ends up being called), we might want to set that up now. Apparently there's a common convention for doing that: https://github.com/pybind/pybind11/issues/1004#issuecomment-322941844 which I think would work in this case as well.

Reply:
I prefer the 2nd proposed naming scheme, i.e. `from moveit_core.kinematic_constraints import utils`.

> If we ever want to add pure python to the package, we might want to set that up now.

I agree with the referenced scheme. I'm using the same approach in MTC.

Reply:
great, than I will implement that and update. If anyone else has thoughts on this please chime in!

Reply:
I'm been implementing this and I have a more specific decision/question... In the exmples I gave, `from moveit_core.kinematic_constraints import utils` the final name `utils` corresponds to a C++ header file. However, I'm wondering if this is unnecessary structure that will make the python code verbose. So keeping with that example, the choice is between:
```
from moveit_core.kinematic_constraints import utils
utils.constructGoalConstraints(...)
from moveit_core.kinematic_constraints.utils import constructGoalConstraints # also works
```
VS
```
from moveit_core import kinematic_constraints
kinematic_constraints.constructGoalConstraints(...)
```


To give another example, let's take the collision_detection namespace. Which do you prefer?
```
from moveit_core.collision_detection import collision_common
from moveit_core.collision_detection import world
world.World(...)
collision_common.CollisionResult(...)
```
VS
```
from moveit_core import collision_detection
collision_detection.World(...)
collision_detection.CollisionResult(...)
```

in C++ obviously the name of the header file doesn't appear in the name of the type, only the include path. So if we want we could drop it, which would make the python docs look more like the C++ docs. If we go the first route, we'd have things like moveit_core.planning_scene.planning_scene.PlanningScene when we could have  moveit_core.planning_scene.PlanningScene. Another advantage is it makes the bindings code shorter/simpler if we drop the distinction between header files and keep it at the namespace level.

Reply:
I prefer the second variant as well. Usually, in C++ we have one class per header file. Hence, there is no need to name the header file once again.